lib/modules/dosomething/dosomething_mbp/dosomething_mbp.admin.inc

```diff
@@ -9,11 +9,212 @@
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for User Registration transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 70,
+      );
+      $template_var = 'dosomething_mbp_user_password_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_user_password_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Password Reset'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Password Reset transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 80,
+      );
+      $template_var = 'dosomething_mbp_campaign_signup_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_campaign_signup_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Campaign Signup'),
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Campaign Signup transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 90,
+      );
+      $template_var = 'dosomething_mbp_campaign_group_signup_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_campaign_group_signup_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Campaign Group Signup'),
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Campaign Group Signup transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 90,
+      );
+      $template_var = 'dosomething_mbp_campaign_reportback_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_campaign_reportback_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Campaign Reportback'),
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Campaign Reportback transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 100,
+      );
+    }
+
+  }
+
   return system_settings_form($form);
 }
+
+/**
+ * Format template data for presentation based on returned values from Mandrill.
+ *
+ * @parm string $application_id
+ *   Defined in message_broker_producer, a unique identifier of the site.
+ */
+function dosomething_mbp_get_mandrill_template_options($application_id) {
```

Comment:
We're also going to lose this function per https://github.com/DoSomething/dosomething/pull/3640#issuecomment-70723782